### PR TITLE
🐛 fix patching project workbench when changing data type 🚨

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- 🐛 fix patching workbench when changing data type (reported by @odeimaiz)
- 🎨 add `--service-opening-waiting-timeout` optional parameter to e2e tests (I committed it by mistake to this branch, but I guess it can go together) 
  - 🚨 modify e2e tests - pointing to a new path 

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by GitHK — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/projects/utils.py [hunk 1]
```diff
@@ -359,7 +359,7 @@ def _cast_outputs_store(dict_data: dict[str, Any]) -> None:
             continue
         # if the entry was modified put the new one
         modified_entry = {k: new_dict[k]}
-        if isinstance(new_dict[k], dict):
+        if isinstance(current_dict[k], dict) and isinstance(new_dict[k], dict):
```
Comment (odeimaiz) [inline services/web/server/src/simcore_service_webserver/projects/utils.py:362]: Thanks 👌

File: services/web/server/tests/unit/with_dbs/02/test_projects_nodes_handlers__patch.py [hunk 2]
```diff
@@ -163,3 +163,62 @@ async def test_patch_project_node(
...
+        "inputs": {
+            "input_3": 0.0,
+            "input_2": 3.0,
+            "input_1": {
+                "store": 0,
+                "path": "api/eddb9098-ac99-331e-930e-d77e25ffe633/file_with_number.txt",
+                "label": "file_with_number.txt",
+                "eTag": "eccbc87e4b5ce2fe28308fd9f2a7baf3",
+                "dataset": None,
+            },
```
Comment (odeimaiz) [inline services/web/server/tests/unit/with_dbs/02/test_projects_nodes_handlers__patch.py:195]: This dict points to a file. Even though I'm aware a node link is also a dict and the test will also pass, I would change it to a link

```suggestion
            "input_1": {
                        "nodeUuid": "c374e5ba-fc42-5c40-ae74-df7ef337f597",
                        "output": "out_1",
            },
```

**Review by sanderegg — APPROVED** [review]
Thanks !

**Review by odeimaiz — APPROVED** [review]
thanks

**Review by bisgaard-itis — APPROVED** [review]
Cool! Thanks for the fix

**Pull request merged**